packages/lesswrong/components/hooks/useRecentOpportunities.ts

```diff
@@ -15,33 +17,73 @@ const subscribedTags: string[] = [
   "ihpwNfh2ZxR4ZHaAK", // Prizes and contests
 ];
 
+type KeyWithSubfield<T, S extends string> = {
+  [K in keyof T]: S extends keyof T[K] ? K : never
+}[keyof T];
+
 export const useRecentOpportunities =<
-  FragmentTypeName extends keyof FragmentTypes
+  FragmentTypeName extends KeyWithSubfield<FragmentTypes, "tagRelevance">
 > ({
   fragmentName,
   limit = 3,
   maxAgeInDays = 7,
+  post,
 }: {
   fragmentName: FragmentTypeName,
   limit?: number,
   maxAgeInDays?: number,
-}): UseMultiResult<FragmentTypeName> => {
+  post?: PostsWithNavigation | PostsWithNavigationAndRevision,
+}): UseMultiResult<FragmentTypeName> & {coreTagLabel: string | null} => {
+  const coreTags =
+    post?.tags
+      ?.filter((tag) => tag.core && !requiredTags.includes(tag._id))
+      .sort((a, b) => (a.name < b.name ? -1 : 1)) ?? [];
+  const coreTagIds = coreTags?.map(tag => tag._id) ?? [];
+  const subscribedTags = difference(defaultSubscribedTags, coreTagIds);
```

Comment:
I'm not sure I understand the purpose of this line? I think you could just use `defaultSubscribedTags` directly below?

Reply:
Oh I see what you mean, sounds good 👍